src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -224,11 +224,62 @@ absl::optional<std::shared_ptr<rpc::GcsNodeInfo>> GcsNodeManager::GetAliveNode(
   return iter->second;
 }
 
-void GcsNodeManager::SetDeathInfo(const NodeID &node_id, rpc::NodeDeathInfo death_info) {
+void GcsNodeManager::SetDeathInfo(const NodeID &node_id,
+                                  shared_ptr<rpc::NodeDeathInfo> death_info) {
   auto maybe_node = GetAliveNode(node_id);
-  RAY_CHECK(maybe_node.has_value());
+  if (!maybe_node.has_value()) {
+    RAY_LOG(INFO) << "Skip setting death info for node " << node_id
+                  << " which is already removed.";
+    return;
+  }
+
+  RAY_CHECK(death_info) << "death_info should not be nullptr";
   auto node = std::move(maybe_node.value());
-  node->mutable_death_info()->CopyFrom(death_info);
+  auto node_death_info = node->mutable_death_info();
+  node_death_info->CopyFrom(*death_info);
+  RAY_LOG(INFO) << "Setting death info for node " << node_id << ": Reason = "
+                << rpc::NodeDeathInfo_Reason_Name(node_death_info->reason())
+                << ", message = " << node_death_info->reason_message();
```

Comment:
Should we print out the death info inside `RemoveNode()`? I think it has an INFO log